localstack/services/s3/provider.py

```diff
@@ -169,8 +202,106 @@ def put_object(
             verify_checksum(checksum_algorithm, context.request.data, request)
 
         response: PutObjectOutput = call_moto(context)
+
+        if expires := request.get("Expires"):
+            moto_backend = get_moto_s3_backend(context)
+            bucket = get_bucket_from_moto(moto_backend, bucket=request.get("Bucket", ""))
+            key = request.get("Key", "")
```

Comment:
i see we're often calling `request.get("Bucket", "")` even for requests where Bucket and Key are required parameters. should we maybe change access to just `request["Bucket"]`? it seems `get_bucket_from_moto` wouldn't work with an empty string anyway, right?

Reply:
I got traumatised by `KeyError` and got too cautious on that one. I'll update that one